web-server/v0.4/e2etests

```diff
@@ -12,16 +17,22 @@ export default {
   },
 };
 EOF
+# Make sure the yarn environment is all up-to-date, or perform the
+# install ahead of actually running the tests.
+printf -- "Running \"yarn install\" command ...\n" 
 yarn --network-timeout 1000000
-if [[ $? -ne 0 ]]; then
+if [[ ${?} -ne 0 ]]; then
   echo "yarn failed!" >&2
   exit 1
 fi
+printf -- "... done running \"yarn install\" command.\n" 
 printf -- "NODE_ENV=%s\n" "${NODE_ENV}"
+printf -- "Running \"yarn start\" command in the background...\n" 
 yarn start &
-if [[ $? -ne 0 ]]; then
-  echo "yarn failed!" >&2
-  exit 1
+yarn_start_pid=${!}
+if [[ -z "${yarn_start_pid}" ]]; then
+    printf -- "yarn e2e tests failed! (could not execute \"yarn start\" in background)\n" >&2
+    exit 1
 fi
 # Wait checking once a second for the web-server to show up on port
 # 8000 at the expected URL.
```

Comment:
Maybe add an explanation of what happens if the curl fails indefinitely?

Reply:
Somebody will have to stop the `e2etest` script and debug why it happened.  In TravisCI the job will just time out and get cleaned up.  The key here is we can't start running the E2E tests until we detect the web server is up and running.  So there is no point doing anything further.

There is a comment below at line 51 what states this.  Is that sufficient?

Reply:
I'd rather have the explanation as a comment, but I won't insist.